documentation/specs/src/interoperability/ethereum-bridge/ethereum_smart_contracts.md

```diff
@@ -1,27 +1,180 @@
+_Bridge_ contract.
+
+The _Proxy_ contract is fixed forever once the bridge has been deployed.
+
+### Bridge
+
+The _Bridge_ contract is the only contract that unprivileged users of the bridge
+may interact with. It provides methods for transferring ERC20s to Namada
+(holding them in escrow in the _Vault_), as well as releasing escrowed ERC20s
+from the _Vault_ for transfers made from Namada to Ethereum. It holds a
+whitelist of ERC20s that may cross the bridge, and this whitelist may be updated
+by the _Governance_ contract.
+
+### Governance
+
+The _Governance_ contract may "upgrade" the bridge by updating the _Proxy_
+contract to point to a new _Bridge_ contract. It may also withdraw all funds
+from the _Vault_ to any specified Ethereum address, if a quorum of validators
+choose to do so.
+
+### wNAM
+
+The _wNAM_ contract is a simple ERC20 token with a fixed supply, which is all
+minted when the bridge is first deployed. After initial deployment, the entire
+supply of _wNAM_ belongs to the _Vault_ contract. As NAM is transferred from
+Namada to Ethereum, wNAM may be released from the _Vault_ by the _Bridge_.
+
+The _wNAM_ contract is fixed forever once the bridge has been deployed.
+
+### Vault
+
+The _Vault_ contract holds in escrow any ERC20 tokens that have been sent over
+the bridge to Namada, as well as a supply of _wNAM_ ERC20s to represent NAM that
+has been sent from Namada to Ethereum. Funds held by the _Vault_ may only be
+spendable by the current _Bridge_ contract. When ERC20 tokens are transferred
+from Ethereum to Namada, they must be deposited to the _Vault_ via the _Bridge_
+contract.
+
+The _Vault_ contract is fixed forever once the bridge has been deployed.
+
+## Namada-side configuration
+
+When an account on Namada becomes a validator, they must provide two Ethereum
+secp256k1 keys:
+
+- a "hot" key for normal operations
+- a "cold" key for exceptional operations, like emergency withdrawal from the
+  bridge
+
+These keys are used to control the bridge smart contracts, via signing of
+messages. Validators may be challenged periodically to prove they still retain
+knowledge of these keys.
+
+## Deployment
+
+The contracts should be deployable by anyone to any EVM chain using an automated
+script. The following configuration should be agreed up front by Namada
+governance before deployment:
+
+- details of the initial active validator set that will control the bridge -
+  specifically, for each validator:
+  - their hot Ethereum address
+  - their cold Ethereum address
+  - their voting power on Namada for the epoch when the bridge will launch
+- the total supply of the wNAM ERC20 token, which will represent Namada-native
+  NAM on the EVM chain
+- an initial whitelist of ERC20 tokens that may cross the bridge from Ethereum
+  to Namada - specifically, for each whitelisted ERC20:
+    - the Ethereum address of the ERC20 contract
+    - a cap on the total amount that may cross the bridge, in units of ERC20
```

Comment:
Not explicitly mentioned here (yet), but the flow rate cap should be the same for every ERC20 (roughly ~5% per day max per ERC20 from Namada to Ethereum)